### ISSUE TITLE
Calling .popover('hide') prevents popover from open on next click

### ISSUE THREAD

**Issue by sbhodge80** [twbs/bootstrap#18860]
Hi,

I think I've found a bug where by if you manually close a popover using the .popover('hide') method, it takes two clicks on the triggering element to show it again: it's as if it thinks the popover is still open and the first click is to  toggle it closed.  I'm entirely prepared to believe this isn't a bug and just that I've misunderstood how it should work!

I've created a fiddle to demonstrate the issue.  The popover contains a button that when clicked, closes the popover using .popover('hide').  When you click the button to show the popover again, nothing happens.  Clicking it a second time then does show the popover.  The fiddle is here:

http://jsfiddle.net/fxqzn4xd/3/
(updated following automated error check)

Thanks
Simon

**Comment by twbs-lmvtfy** [comment]
Hi @sbhodge80!

You appear to have posted a live example (http://fiddle.jshell.net/fxqzn4xd/1/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 84, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 31, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.6 ; saw what appears to be usage of Bootstrap 3.3.5
* line 35, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.6 ; saw what appears to be usage of Bootstrap 3.3.5

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by sbhodge80** [comment]
I have corrected the errors.  The revised example is here:

http://jsfiddle.net/fxqzn4xd/3/

Thanks
Simon

**Comment by cvrebert** [comment]
Duplicate of #16732.

**Comment by SamanChalian** [comment]
> Duplicate of #16732.

Instead of answering to the question just looking for the duplication with a broken link!!! I'm sorry for for you because of this way of thinking.

**Comment by SamanChalian** [comment]
> I have corrected the errors. The revised example is here:
> 
> http://jsfiddle.net/fxqzn4xd/3/
> 
> Thanks
> Simon

Could you find a way to solve that issue?

**Comment by KateHodge** [comment]
Hi, yes, I can't remember exactly what led me to this so can't credit it, but using the following code hooks into the hidden event to reset the popover's click state, which doesn't happen when you manually hide it:

```
        $('body').on('hidden.bs.popover', function (e) {
            var popoverData = $(e.target).data("bs.popover");
            if (popoverData) {
                popoverData.inState.click = false;
            }
        });
```

**Comment by SamanChalian** [comment]
> $('body').on('hidden.bs.popover', function (e) {
>             var popoverData = $(e.target).data("bs.popover");
>             if (popoverData) {
>                 popoverData.inState.click = false;
>             }
>         });

Hi Kate, Thank you so much. That works nice. 👍
